testing/sample_framework/src/ios/ios_app_framework.mm

```diff
@@ -87,9 +101,20 @@ - (void)viewDidLoad {
 namespace app_framework {
 
 bool ProcessEvents(int msec) {
-  [g_shutdown_signal
-      waitUntilDate:[NSDate dateWithTimeIntervalSinceNow:static_cast<float>(msec) / 1000.0f]];
-  return g_shutdown;
+  NSDate* endDate = [NSDate dateWithTimeIntervalSinceNow:static_cast<float>(msec) / 1000.0f];
+  [g_running_status_condition lock];
+
+  if (g_running_status == RunningStatus::kRunning) {
+    NSDate* currentDate = [NSDate date];
+    if ([currentDate compare:endDate] == NSOrderedAscending) {
```

Comment:
Question: what is the reason to check this manually? Is there any advantage to calling `wait` unconditionally and letting its implementation figure out if enough time has passed?